components/membership/synchronisation-impl/src/integrationTest/kotlin/net/corda/membership/impl/synchronisation/MemberSynchronisationIntegrationTest.kt

```diff
@@ -0,0 +1,378 @@
+                    AppMessage(
+                        AuthenticatedMessage(
+                            messageHeader,
+                            payload
+                        )
+                    )
+                )
+            )
+        )
+
+        // Receive and assert results
+        val result = assertDoesNotThrow {
+            completableResult.getOrThrow(Duration.ofSeconds(5))
+        }
+        registrationRequestSubscription.close()
+
+        SoftAssertions.assertSoftly {
+            it.assertThat(result).isNotNull
+            it.assertThat(result?.second)
+                .isNotNull
+                .isInstanceOf(PersistentMemberInfo::class.java)
+            with(result!!.second) {
+                it.assertThat(viewOwningMember).isEqualTo(destination)
+                val memberPublished = memberInfoFactory.create(
+                    memberContext.toSortedMap(),
+                    mgmContext.toSortedMap()
+                )
+                it.assertThat(memberPublished.groupId).isEqualTo(groupId)
+                it.assertThat(memberPublished.name.toString()).isEqualTo(participant.x500Name)
+                it.assertThat(memberPublished.ledgerKeys.size).isEqualTo(0)
+                it.assertThat(memberPublished.status).isEqualTo(MEMBER_STATUS_ACTIVE)
+                it.assertThat(memberPublished.modifiedTime).isEqualTo(time)
+            }
+        }
+    }
+
+    @Suppress("SpreadOperator")
+    private fun createTestMemberInfo(holdingIdentity: HoldingIdentity): MemberInfo = memberInfoFactory.create(
+        sortedMapOf(
+            MemberInfoExtension.PARTY_NAME to holdingIdentity.x500Name,
+            MemberInfoExtension.PARTY_SESSION_KEY to "dummy-session-key",
+            MemberInfoExtension.GROUP_ID to groupId,
+            *convertEndpoints().toTypedArray(),
+            MemberInfoExtension.SOFTWARE_VERSION to "5.0.0",
+            MemberInfoExtension.PLATFORM_VERSION to "10",
+            MemberInfoExtension.SERIAL to "1",
+        ),
+        sortedMapOf(
+            MemberInfoExtension.STATUS to MEMBER_STATUS_ACTIVE,
+            MemberInfoExtension.MODIFIED_TIME to clock.instant().toString(),
+        )
+
+    )
+
+    private fun convertEndpoints(): List<Pair<String, String>> {
+        val result = mutableListOf<Pair<String, String>>()
+        for (index in endpoints.indices) {
+            result.add(
+                Pair(
+                    String.format(MemberInfoExtension.URL_KEY, index),
+                    endpoints[index].url
+                )
+            )
+            result.add(
+                Pair(
+                    String.format(MemberInfoExtension.PROTOCOL_VERSION, index),
+                    endpoints[index].protocolVersion.toString()
+                )
+            )
+        }
+        return result
+    }
+
+    private fun getTestProcessor(resultCollector: (String, PersistentMemberInfo) -> Unit): PubSubProcessor<String, PersistentMemberInfo> {
```

Comment:
Could potentially change this to 
```suggestion
    private fun getTestProcessor(resultCollector: (PersistentMemberInfo) -> Unit): PubSubProcessor<String, PersistentMemberInfo> {
```
And below have `resultCollector(event.value!!)` since the assertions above are only on the value